apps/meteor/client/views/marketplace/hooks/useAppRequests.ts

```diff
@@ -0,0 +1,14 @@
+import type { AppRequestFilter } from '@rocket.chat/core-typings';
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+export const useAppRequests = (appId: string, limit?: number, offset?: number, sort?: string, filter?: AppRequestFilter) => {
+	const fetchAppRequests = useEndpoint('GET', '/apps/app-request');
+
+	return useQuery({
+		queryKey: ['app-requests', limit, offset],
```

Comment:
is that correct? what happens if the appid ,limit changes?

Reply:
This was misunderstood of how tanstack query deals with the cache. Also this was done because the pagination of the App requests tab is done in the back-end so it needs to reload the data if the user changes the local items per page or page number of the pagination.